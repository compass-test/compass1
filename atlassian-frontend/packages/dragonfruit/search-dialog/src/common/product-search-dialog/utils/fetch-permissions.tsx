import { Products } from '../../../common/product-context';
import { SharedClient } from '../../clients';

export const fetchPermissions = (
  sharedClient: SharedClient,
  products: Products[],
) => {
  // For MVP, it is assumed the user has permissions to the primary product
  // and ignoring the returned scopes
  if (products.length === 0 || products.length === 1) {
    return Promise.resolve(products);
  }

  return sharedClient
    .getProductPermissions(products.slice(1, products.length))
    .then((permissibleProducts) => [products[0], ...permissibleProducts])
    .catch(() => {
      return products.slice(0, 1);
    });
};
