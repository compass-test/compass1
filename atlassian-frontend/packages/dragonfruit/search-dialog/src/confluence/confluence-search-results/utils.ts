import { User } from '@atlaskit/user-picker';
import { routes } from '@atlassian/dragonfruit-routes';
import type { RecentlyViewedTeam } from '@atlassian/compass-search-cache';

export const isUser = (
  teamOrUser: RecentlyViewedTeam | User,
): teamOrUser is User => (teamOrUser as User).name !== undefined;

export const getTeamOrUserDisplayName = (
  teamOrUser: RecentlyViewedTeam | User,
): string => (isUser(teamOrUser) ? teamOrUser.name : teamOrUser.displayName);

export const getTeamOrUserAvatar = (
  teamOrUser: RecentlyViewedTeam | User,
): string | undefined =>
  isUser(teamOrUser) ? teamOrUser.avatarUrl : teamOrUser.smallAvatarImageUrl;

export const getTeamOrUserURL = (
  teamOrUser: RecentlyViewedTeam | User,
): string =>
  isUser(teamOrUser)
    ? `${window.location.origin}/people/${teamOrUser.id}`
    : routes.TEAM_DETAILS(teamOrUser.id.replace('ari:cloud:teams::team/', ''));
