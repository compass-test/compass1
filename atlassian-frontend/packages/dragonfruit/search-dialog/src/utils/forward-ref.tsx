/**
 * Thin helper method to simplify forward ref
 */
import React from 'react';

export const withForwardRef = <
  T extends HTMLElement,
  P extends { forwardRef: React.Ref<T> }
>(
  Component: React.ComponentType<P>,
) => {
  return React.forwardRef<T, P>((props, ref) => (
    <Component {...props} forwardRef={ref} />
  )) as React.ForwardRefExoticComponent<
    Omit<P, 'forwardRef'> & React.RefAttributes<T>
  >;
};
