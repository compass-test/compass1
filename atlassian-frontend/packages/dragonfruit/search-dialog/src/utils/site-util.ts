// eslint-disable-next-line import/no-extraneous-dependencies
import faker from 'faker';
import { Products } from '../';

export const generateMockSites = (numSites: number, product: Products) =>
  Array(numSites)
    .fill(0)
    .map(() => ({
      product: product,
      avatarUrl: 'https://i.imgur.com/0YmmsuX.jpg',
      cloudId: faker.random.uuid(),
      siteName: `${faker.random.word()}`,
      siteUrl: `https://${faker.random.word()}.atlassian.net`,
    }));

export const generateMockSitesWithAdvancedSearch = (
  numSites: number,
  product: Products,
) =>
  generateMockSites(numSites, product).map((s) => ({
    ...s,
    advancedSearchUrl: faker.internet.url(),
  }));
