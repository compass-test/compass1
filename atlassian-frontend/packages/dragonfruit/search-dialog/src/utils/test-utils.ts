export function nextTickAfterDebouncedFunctionExecution() {
  return new Promise((resolve) => setTimeout(resolve, 300));
}
