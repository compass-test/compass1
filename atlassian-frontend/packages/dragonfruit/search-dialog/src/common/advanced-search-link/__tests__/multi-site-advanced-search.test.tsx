import React from 'react';
import MultiSiteAdvancedSearchLink, {
  Props,
} from '../multi-site-advanced-search-link';
import { mountWithIntl } from '../../../__tests__/__fixtures__/intl-test-helpers';
import { Products } from '../../../';
import { generateMockSites } from '../../../utils/site-util';

const mockFireEvent = jest.fn();

const MOCK_SITES = generateMockSites(1, Products.confluence);

jest.mock('../../analytics', () =>
  Object.assign({}, jest.requireActual('../../analytics'), {
    useAnalytics: () => ({
      fireAnalyticsEvent: mockFireEvent,
    }),
  }),
);

describe('<MultiSiteAdvancedSearch />', () => {
  const defaultProps: Props = {
    sites: MOCK_SITES.map((s) => ({
      ...s,
      advancedSearchUrl: 'some-advanced-search-url',
    })),
    isLoading: false,
  };

  it('MultiSiteAdvancedSearch matches snapshot', () => {
    const wrapper = mountWithIntl(
      <MultiSiteAdvancedSearchLink {...defaultProps} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('MultiSiteAdvancedSearch matches snapshot with custom component', () => {
    const wrapper = mountWithIntl(
      <MultiSiteAdvancedSearchLink
        {...defaultProps}
        linkComponent={(props) => <div {...props} />}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('calls onclick', () => {
    const mockOnClick = jest.fn();
    const wrapper = mountWithIntl(
      <MultiSiteAdvancedSearchLink
        {...defaultProps}
        onClick={mockOnClick}
        linkComponent={(props) => <div {...props} />}
      />,
    );

    wrapper.find('button').simulate('click');
    expect(true).toEqual(true);

    expect(mockOnClick).toHaveBeenCalledWith(
      'some-advanced-search-url',
      expect.anything(),
    );
  });

  it('uses a link component', () => {
    const mockOnClick = jest.fn();
    const wrapper = mountWithIntl(
      <MultiSiteAdvancedSearchLink
        {...defaultProps}
        linkComponent={(props) => <div {...props} onClick={mockOnClick} />}
      />,
    );

    wrapper.find('button').simulate('click');
    expect(true).toEqual(true);

    expect(mockOnClick).toHaveBeenCalled();
  });

  it('fires analytics when clicked', () => {
    const wrapper = mountWithIntl(
      <MultiSiteAdvancedSearchLink
        {...defaultProps}
        linkComponent={(props) => <div {...props} />}
      />,
    );

    wrapper.find('button').simulate('click');

    expect(mockFireEvent).toHaveBeenCalledWith({
      action: 'selected',
      actionSubject: 'advancedSearchLink',
      actionSubjectId: 'confluenceAdvancedSearchLink',
      attributes: {
        destinationId: MOCK_SITES[0].cloudId,
        index: 0,
        isLoading: false,
        newTab: false,
        trigger: 'click',
      },
      eventType: 'ui',
      source: 'searchDialog',
    });
  });
});
