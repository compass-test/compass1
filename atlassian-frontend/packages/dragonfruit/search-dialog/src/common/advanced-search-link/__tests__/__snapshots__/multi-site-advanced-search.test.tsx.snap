// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultiSiteAdvancedSearch /> MultiSiteAdvancedSearch matches snapshot 1`] = `
.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 12px 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 780px;
  color: #091E42;
}

.emotion-0 {
  margin-right: 4px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-6 {
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
  overflow: hidden;
  max-width: -webkit-max-content;
  max-width: -moz-max-content;
  max-width: max-content;
  margin-right: 2px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-overflow: ellipsis;
  border-radius: 3px;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  -webkit-transition: box-shadow 0.15s cubic-bezier(0.47,0.03,0.49,1.38);
  transition: box-shadow 0.15s cubic-bezier(0.47,0.03,0.49,1.38);
}

.emotion-4:focus {
  outline: none;
  box-shadow: 0 0 0 2px #4C9AFF inset;
}

.emotion-4:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3 {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-2 {
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  border-width: 0;
  border-radius: 3px;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: inherit;
  font-style: normal;
  font-family: inherit;
  font-weight: 500;
  max-width: 100%;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background 0.1s ease-out,box-shadow 0.15s cubic-bezier(0.47,0.03,0.49,1.38);
  transition: background 0.1s ease-out,box-shadow 0.15s cubic-bezier(0.47,0.03,0.49,1.38);
  white-space: nowrap;
  background: var(--background-subtleNeutral-resting,rgba(9,30,66,0.04));
  color: var(--text-highEmphasis,#42526E) !important;
  cursor: pointer;
  height: 1.7142857142857142em;
  line-height: 1.7142857142857142em;
  padding: 0 10px;
  vertical-align: middle;
  width: auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-2:visited {
  background: var(--background-subtleNeutral-resting,rgba(9,30,66,0.04));
  color: var(--text-highEmphasis,#42526E) !important;
}

.emotion-2:hover {
  background: var(--background-subtleNeutral-hover,rgba(9,30,66,0.08));
  color: var(--text-highEmphasis,#42526E) !important;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  -webkit-transition-duration: 0s,0.15s;
  transition-duration: 0s,0.15s;
}

.emotion-2:focus {
  background: var(--background-subtleNeutral-resting,rgba(9,30,66,0.04));
  color: var(--text-highEmphasis,#42526E) !important;
  box-shadow: 0 0 0 2px var(--border-focus,#4C9AFF);
  -webkit-transition-duration: 0s,0.2s;
  transition-duration: 0s,0.2s;
  outline: none;
}

.emotion-2:active {
  background: var(--background-subtleNeutral-pressed,rgba(179,212,255,0.6));
  color: var(--text-highEmphasis,#0052CC) !important;
  -webkit-transition-duration: 0s,0s;
  transition-duration: 0s,0s;
}

.emotion-2[data-firefox-is-active="true"] {
  background: var(--background-subtleNeutral-pressed,rgba(179,212,255,0.6));
  color: var(--text-highEmphasis,#0052CC) !important;
  -webkit-transition-duration: 0s,0s;
  transition-duration: 0s,0s;
}

.emotion-2[disabled] {
  background: var(--background-disabled,rgba(9,30,66,0.04));
  color: var(--text-disabled,#A5ADBA) !important;
  cursor: not-allowed;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2[data-has-overlay="true"] {
  cursor: default;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2[data-has-overlay="true"]:not([disabled]):hover,
.emotion-2[data-has-overlay="true"]:not([disabled]):active {
  background: var(--background-subtleNeutral-resting,rgba(9,30,66,0.04));
  color: var(--text-highEmphasis,#42526E) !important;
}

.emotion-2::-moz-focus-inner {
  border: 0;
  margin: 0;
  padding: 0;
}

.emotion-1 {
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
  opacity: 1;
  margin: 0 2px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<MultiSiteAdvancedSearchLink
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  isLoading={false}
  sites={
    Array [
      Object {
        "advancedSearchUrl": "some-advanced-search-url",
        "avatarUrl": "https://i.imgur.com/0YmmsuX.jpg",
        "cloudId": "04c60c34-40ba-4ca3-ba25-36cd8e23c4c5",
        "product": "CONFLUENCE",
        "siteName": "Investor",
        "siteUrl": "https://generating.atlassian.net",
      },
    ]
  }
>
  <SearchFooterLinks
    dropdownTriggerLabel="-4 more"
    label="Search on:"
    links={
      Array [
        Object {
          "content": "Investor",
          "href": "some-advanced-search-url",
          "key": "04c60c34-40ba-4ca3-ba25-36cd8e23c4c5",
        },
      ]
    }
    maxPrimaryLinks={5}
    onClick={[Function]}
  >
    <Styled(div)>
      <div
        className="emotion-7"
      >
        <Styled(div)>
          <div
            className="emotion-0"
          >
            Search on:
          </div>
        </Styled(div)>
        <Styled(div)
          key="04c60c34-40ba-4ca3-ba25-36cd8e23c4c5"
          onClick={[Function]}
        >
          <div
            className="emotion-6"
            onClick={[Function]}
          >
            <Styled(Link)
              href="some-advanced-search-url"
            >
              <Link
                className="emotion-4"
                href="some-advanced-search-url"
              >
                <a
                  className="emotion-4"
                  href="some-advanced-search-url"
                >
                  <Tooltip
                    content="Investor"
                    tag={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "render": [Function],
                      }
                    }
                  >
                    <ForwardRef
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onMouseDown={[Function]}
                      onMouseOut={[Function]}
                      onMouseOver={[Function]}
                      role="presentation"
                    >
                      <Styled(div)
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onMouseDown={[Function]}
                        onMouseOut={[Function]}
                        onMouseOver={[Function]}
                        role="presentation"
                      >
                        <div
                          className="emotion-3"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onMouseDown={[Function]}
                          onMouseOut={[Function]}
                          onMouseOver={[Function]}
                          role="presentation"
                        >
                          <ForwardRef(Button)
                            spacing="compact"
                            tabIndex={-1}
                          >
                            <Consumer>
                              <ForwardRef(ButtonWithMode)
                                mode="light"
                                spacing="compact"
                                tabIndex={-1}
                              >
                                <ForwardRef(ButtonBase)
                                  buttonCss={
                                    Object {
                                      "&::-moz-focus-inner": Object {
                                        "border": 0,
                                        "margin": 0,
                                        "padding": 0,
                                      },
                                      "&:active": Object {
                                        "background": "var(--background-subtleNeutral-pressed, rgba(179, 212, 255, 0.6))",
                                        "color": "var(--text-highEmphasis, #0052CC) !important",
                                        "transitionDuration": "0s, 0s",
                                      },
                                      "&:focus": Object {
                                        "background": "var(--background-subtleNeutral-resting, rgba(9, 30, 66, 0.04))",
                                        "boxShadow": "0 0 0 2px var(--border-focus, #4C9AFF)",
                                        "color": "var(--text-highEmphasis, #42526E) !important",
                                        "outline": "none",
                                        "transitionDuration": "0s, 0.2s",
                                      },
                                      "&:hover": Object {
                                        "background": "var(--background-subtleNeutral-hover, rgba(9, 30, 66, 0.08))",
                                        "color": "var(--text-highEmphasis, #42526E) !important",
                                        "textDecoration": "inherit",
                                        "transitionDuration": "0s, 0.15s",
                                      },
                                      "&:visited": Object {
                                        "background": "var(--background-subtleNeutral-resting, rgba(9, 30, 66, 0.04))",
                                        "color": "var(--text-highEmphasis, #42526E) !important",
                                      },
                                      "&[data-firefox-is-active=\\"true\\"]": Object {
                                        "background": "var(--background-subtleNeutral-pressed, rgba(179, 212, 255, 0.6))",
                                        "color": "var(--text-highEmphasis, #0052CC) !important",
                                        "transitionDuration": "0s, 0s",
                                      },
                                      "&[data-has-overlay=\\"true\\"]": Object {
                                        "cursor": "default",
                                        "textDecoration": "none",
                                      },
                                      "&[data-has-overlay=\\"true\\"]:not([disabled]):hover, &[data-has-overlay=\\"true\\"]:not([disabled]):active": Object {
                                        "background": "var(--background-subtleNeutral-resting, rgba(9, 30, 66, 0.04))",
                                        "color": "var(--text-highEmphasis, #42526E) !important",
                                      },
                                      "&[disabled]": Object {
                                        "background": "var(--background-disabled, rgba(9, 30, 66, 0.04))",
                                        "color": "var(--text-disabled, #A5ADBA) !important",
                                        "cursor": "not-allowed",
                                        "textDecoration": "none",
                                      },
                                      "alignItems": "baseline",
                                      "background": "var(--background-subtleNeutral-resting, rgba(9, 30, 66, 0.04))",
                                      "borderRadius": 3,
                                      "borderWidth": 0,
                                      "boxSizing": "border-box",
                                      "color": "var(--text-highEmphasis, #42526E) !important",
                                      "cursor": "pointer",
                                      "display": "inline-flex",
                                      "fontFamily": "inherit",
                                      "fontSize": "inherit",
                                      "fontStyle": "normal",
                                      "fontWeight": 500,
                                      "height": "1.7142857142857142em",
                                      "justifyContent": "center",
                                      "lineHeight": "1.7142857142857142em",
                                      "maxWidth": "100%",
                                      "padding": "0 10px",
                                      "position": "relative",
                                      "textAlign": "center",
                                      "textDecoration": "none",
                                      "transition": "background 0.1s ease-out, box-shadow 0.15s cubic-bezier(0.47, 0.03, 0.49, 1.38)",
                                      "verticalAlign": "middle",
                                      "whiteSpace": "nowrap",
                                      "width": "auto",
                                    }
                                  }
                                  onMouseDown={[Function]}
                                  onMouseUp={[Function]}
                                  spacing="compact"
                                  tabIndex={-1}
                                >
                                  <button
                                    className="emotion-2"
                                    disabled={false}
                                    onClick={[Function]}
                                    onMouseDown={[Function]}
                                    onMouseUp={[Function]}
                                    tabIndex={-1}
                                    type="button"
                                  >
                                    <span
                                      className="emotion-1"
                                    >
                                      Investor
                                    </span>
                                  </button>
                                </ForwardRef(ButtonBase)>
                              </ForwardRef(ButtonWithMode)>
                            </Consumer>
                          </ForwardRef(Button)>
                        </div>
                      </Styled(div)>
                    </ForwardRef>
                  </Tooltip>
                </a>
              </Link>
            </Styled(Link)>
          </div>
        </Styled(div)>
      </div>
    </Styled(div)>
  </SearchFooterLinks>
</MultiSiteAdvancedSearchLink>
`;

exports[`<MultiSiteAdvancedSearch /> MultiSiteAdvancedSearch matches snapshot with custom component 1`] = `
.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 12px 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 780px;
  color: #091E42;
}

.emotion-0 {
  margin-right: 4px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-7 {
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
  overflow: hidden;
  max-width: -webkit-max-content;
  max-width: -moz-max-content;
  max-width: max-content;
  margin-right: 2px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-overflow: ellipsis;
  border-radius: 3px;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  -webkit-transition: box-shadow 0.15s cubic-bezier(0.47,0.03,0.49,1.38);
  transition: box-shadow 0.15s cubic-bezier(0.47,0.03,0.49,1.38);
}

.emotion-4:focus {
  outline: none;
  box-shadow: 0 0 0 2px #4C9AFF inset;
}

.emotion-4:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3 {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-2 {
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  border-width: 0;
  border-radius: 3px;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: inherit;
  font-style: normal;
  font-family: inherit;
  font-weight: 500;
  max-width: 100%;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background 0.1s ease-out,box-shadow 0.15s cubic-bezier(0.47,0.03,0.49,1.38);
  transition: background 0.1s ease-out,box-shadow 0.15s cubic-bezier(0.47,0.03,0.49,1.38);
  white-space: nowrap;
  background: var(--background-subtleNeutral-resting,rgba(9,30,66,0.04));
  color: var(--text-highEmphasis,#42526E) !important;
  cursor: pointer;
  height: 1.7142857142857142em;
  line-height: 1.7142857142857142em;
  padding: 0 10px;
  vertical-align: middle;
  width: auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-2:visited {
  background: var(--background-subtleNeutral-resting,rgba(9,30,66,0.04));
  color: var(--text-highEmphasis,#42526E) !important;
}

.emotion-2:hover {
  background: var(--background-subtleNeutral-hover,rgba(9,30,66,0.08));
  color: var(--text-highEmphasis,#42526E) !important;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  -webkit-transition-duration: 0s,0.15s;
  transition-duration: 0s,0.15s;
}

.emotion-2:focus {
  background: var(--background-subtleNeutral-resting,rgba(9,30,66,0.04));
  color: var(--text-highEmphasis,#42526E) !important;
  box-shadow: 0 0 0 2px var(--border-focus,#4C9AFF);
  -webkit-transition-duration: 0s,0.2s;
  transition-duration: 0s,0.2s;
  outline: none;
}

.emotion-2:active {
  background: var(--background-subtleNeutral-pressed,rgba(179,212,255,0.6));
  color: var(--text-highEmphasis,#0052CC) !important;
  -webkit-transition-duration: 0s,0s;
  transition-duration: 0s,0s;
}

.emotion-2[data-firefox-is-active="true"] {
  background: var(--background-subtleNeutral-pressed,rgba(179,212,255,0.6));
  color: var(--text-highEmphasis,#0052CC) !important;
  -webkit-transition-duration: 0s,0s;
  transition-duration: 0s,0s;
}

.emotion-2[disabled] {
  background: var(--background-disabled,rgba(9,30,66,0.04));
  color: var(--text-disabled,#A5ADBA) !important;
  cursor: not-allowed;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2[data-has-overlay="true"] {
  cursor: default;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2[data-has-overlay="true"]:not([disabled]):hover,
.emotion-2[data-has-overlay="true"]:not([disabled]):active {
  background: var(--background-subtleNeutral-resting,rgba(9,30,66,0.04));
  color: var(--text-highEmphasis,#42526E) !important;
}

.emotion-2::-moz-focus-inner {
  border: 0;
  margin: 0;
  padding: 0;
}

.emotion-1 {
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
  opacity: 1;
  margin: 0 2px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<MultiSiteAdvancedSearchLink
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  isLoading={false}
  linkComponent={[Function]}
  sites={
    Array [
      Object {
        "advancedSearchUrl": "some-advanced-search-url",
        "avatarUrl": "https://i.imgur.com/0YmmsuX.jpg",
        "cloudId": "04c60c34-40ba-4ca3-ba25-36cd8e23c4c5",
        "product": "CONFLUENCE",
        "siteName": "Investor",
        "siteUrl": "https://generating.atlassian.net",
      },
    ]
  }
>
  <SearchFooterLinks
    dropdownTriggerLabel="-4 more"
    label="Search on:"
    linkComponent={[Function]}
    links={
      Array [
        Object {
          "content": "Investor",
          "href": "some-advanced-search-url",
          "key": "04c60c34-40ba-4ca3-ba25-36cd8e23c4c5",
        },
      ]
    }
    maxPrimaryLinks={5}
    onClick={[Function]}
  >
    <Styled(div)>
      <div
        className="emotion-8"
      >
        <Styled(div)>
          <div
            className="emotion-0"
          >
            Search on:
          </div>
        </Styled(div)>
        <Styled(div)
          key="04c60c34-40ba-4ca3-ba25-36cd8e23c4c5"
          onClick={[Function]}
        >
          <div
            className="emotion-7"
            onClick={[Function]}
          >
            <Styled(Link)
              href="some-advanced-search-url"
              linkComponent={[Function]}
            >
              <Link
                className="emotion-4"
                href="some-advanced-search-url"
                linkComponent={[Function]}
              >
                <linkComponent
                  className="emotion-4"
                  href="some-advanced-search-url"
                >
                  <div
                    className="emotion-4"
                    href="some-advanced-search-url"
                  >
                    <Tooltip
                      content="Investor"
                      tag={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        }
                      }
                    >
                      <ForwardRef
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onMouseDown={[Function]}
                        onMouseOut={[Function]}
                        onMouseOver={[Function]}
                        role="presentation"
                      >
                        <Styled(div)
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onMouseDown={[Function]}
                          onMouseOut={[Function]}
                          onMouseOver={[Function]}
                          role="presentation"
                        >
                          <div
                            className="emotion-3"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onMouseDown={[Function]}
                            onMouseOut={[Function]}
                            onMouseOver={[Function]}
                            role="presentation"
                          >
                            <ForwardRef(Button)
                              spacing="compact"
                              tabIndex={-1}
                            >
                              <Consumer>
                                <ForwardRef(ButtonWithMode)
                                  mode="light"
                                  spacing="compact"
                                  tabIndex={-1}
                                >
                                  <ForwardRef(ButtonBase)
                                    buttonCss={
                                      Object {
                                        "&::-moz-focus-inner": Object {
                                          "border": 0,
                                          "margin": 0,
                                          "padding": 0,
                                        },
                                        "&:active": Object {
                                          "background": "var(--background-subtleNeutral-pressed, rgba(179, 212, 255, 0.6))",
                                          "color": "var(--text-highEmphasis, #0052CC) !important",
                                          "transitionDuration": "0s, 0s",
                                        },
                                        "&:focus": Object {
                                          "background": "var(--background-subtleNeutral-resting, rgba(9, 30, 66, 0.04))",
                                          "boxShadow": "0 0 0 2px var(--border-focus, #4C9AFF)",
                                          "color": "var(--text-highEmphasis, #42526E) !important",
                                          "outline": "none",
                                          "transitionDuration": "0s, 0.2s",
                                        },
                                        "&:hover": Object {
                                          "background": "var(--background-subtleNeutral-hover, rgba(9, 30, 66, 0.08))",
                                          "color": "var(--text-highEmphasis, #42526E) !important",
                                          "textDecoration": "inherit",
                                          "transitionDuration": "0s, 0.15s",
                                        },
                                        "&:visited": Object {
                                          "background": "var(--background-subtleNeutral-resting, rgba(9, 30, 66, 0.04))",
                                          "color": "var(--text-highEmphasis, #42526E) !important",
                                        },
                                        "&[data-firefox-is-active=\\"true\\"]": Object {
                                          "background": "var(--background-subtleNeutral-pressed, rgba(179, 212, 255, 0.6))",
                                          "color": "var(--text-highEmphasis, #0052CC) !important",
                                          "transitionDuration": "0s, 0s",
                                        },
                                        "&[data-has-overlay=\\"true\\"]": Object {
                                          "cursor": "default",
                                          "textDecoration": "none",
                                        },
                                        "&[data-has-overlay=\\"true\\"]:not([disabled]):hover, &[data-has-overlay=\\"true\\"]:not([disabled]):active": Object {
                                          "background": "var(--background-subtleNeutral-resting, rgba(9, 30, 66, 0.04))",
                                          "color": "var(--text-highEmphasis, #42526E) !important",
                                        },
                                        "&[disabled]": Object {
                                          "background": "var(--background-disabled, rgba(9, 30, 66, 0.04))",
                                          "color": "var(--text-disabled, #A5ADBA) !important",
                                          "cursor": "not-allowed",
                                          "textDecoration": "none",
                                        },
                                        "alignItems": "baseline",
                                        "background": "var(--background-subtleNeutral-resting, rgba(9, 30, 66, 0.04))",
                                        "borderRadius": 3,
                                        "borderWidth": 0,
                                        "boxSizing": "border-box",
                                        "color": "var(--text-highEmphasis, #42526E) !important",
                                        "cursor": "pointer",
                                        "display": "inline-flex",
                                        "fontFamily": "inherit",
                                        "fontSize": "inherit",
                                        "fontStyle": "normal",
                                        "fontWeight": 500,
                                        "height": "1.7142857142857142em",
                                        "justifyContent": "center",
                                        "lineHeight": "1.7142857142857142em",
                                        "maxWidth": "100%",
                                        "padding": "0 10px",
                                        "position": "relative",
                                        "textAlign": "center",
                                        "textDecoration": "none",
                                        "transition": "background 0.1s ease-out, box-shadow 0.15s cubic-bezier(0.47, 0.03, 0.49, 1.38)",
                                        "verticalAlign": "middle",
                                        "whiteSpace": "nowrap",
                                        "width": "auto",
                                      }
                                    }
                                    onMouseDown={[Function]}
                                    onMouseUp={[Function]}
                                    spacing="compact"
                                    tabIndex={-1}
                                  >
                                    <button
                                      className="emotion-2"
                                      disabled={false}
                                      onClick={[Function]}
                                      onMouseDown={[Function]}
                                      onMouseUp={[Function]}
                                      tabIndex={-1}
                                      type="button"
                                    >
                                      <span
                                        className="emotion-1"
                                      >
                                        Investor
                                      </span>
                                    </button>
                                  </ForwardRef(ButtonBase)>
                                </ForwardRef(ButtonWithMode)>
                              </Consumer>
                            </ForwardRef(Button)>
                          </div>
                        </Styled(div)>
                      </ForwardRef>
                    </Tooltip>
                  </div>
                </linkComponent>
              </Link>
            </Styled(Link)>
          </div>
        </Styled(div)>
      </div>
    </Styled(div)>
  </SearchFooterLinks>
</MultiSiteAdvancedSearchLink>
`;
